### PR TITLE
feat(ONYX-1289): add tracking to news section

### PR DESCRIPTION
**Pull request by MounirDhahri**
This PR resolves [ONYX-1289] 

### Description

This PR makes the following changes:
- Add tracking to news articles taps
- I noticed that the navigation to the news article was broken - we were navigating `/articles` instead of `article.href`. I fixed that.

This PR is blocked by this cohesion PR: https://github.com/artsy/cohesion/pull/526

### PR Checklist

- [x] I have tested my changes on **iOS** and **Android**.
- [x] I hid my changes behind a **[feature flag]**, or they don't need one.
- [x] I have included **screenshots** or **videos**, or I have not changed the UI.
- [x] I have added **tests**, or my changes don't require any.
- [x] I added an **[app state migration]**, or my changes do not require one.
- [x] I have documented any **follow-up work** that this PR will require, or it does not require any.
- [x] I have added a **changelog entry** below, or my changes do not require one.

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to **run** this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

- add tracking to the news home view section - mounir

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

-

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

[ONYX-1289]: https://artsyproduct.atlassian.net/browse/ONYX-1289?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by nickskalkin — APPROVED** [review]

**Review by dblandin — CHANGES_REQUESTED** [review]
I added a comment [here in cohesion](https://github.com/artsy/cohesion/pull/526#discussion_r1756503321). To reiterate, I wonder if we can use the existing [tappedArticleGroup](https://github.com/artsy/cohesion/blob/a99edf3eb623dbf77dc7a1f839f1a2dd934d4c95/src/Schema/Events/Tap.ts#L12-L34) event. Seems to operate in an identical way.

File: src/app/Scenes/HomeView/Sections/HomeViewSectionArticlesCards.tsx [hunk 1]
```diff
@@ -34,7 +32,12 @@ export const HomeViewSectionArticlesCards: React.FC<HomeViewSectionArticlesCards
       </Flex>
       {articles.map((article, index) => (
         <Flex key={index} gap={space(2)}>
-          <Touchable onPress={handleOnPress}>
+          <Touchable
+            onPress={() => {
+              tracking.tappedNewsGroup(article.internalID, article.slug, section.internalID, index)
+              navigate(article.href ?? "")
```
[the remark this answers is not in the document]
Comment (anandaroop) [reply src/app/Scenes/HomeView/Sections/HomeViewSectionArticlesCards.tsx:38]: **question (non-blocking):** Maybe edge case, but what _is_ the expected behavior if you were to pass in "" to `navigate`? Should we guard against that in the first place by only navigating if `article.href` is present?

File: src/app/Scenes/HomeView/Sections/HomeViewSectionArticlesCards.tsx [hunk 2]
```diff
@@ -34,7 +32,12 @@ export const HomeViewSectionArticlesCards: React.FC<HomeViewSectionArticlesCards
       </Flex>
       {articles.map((article, index) => (
         <Flex key={index} gap={space(2)}>
-          <Touchable onPress={handleOnPress}>
+          <Touchable
+            onPress={() => {
+              tracking.tappedNewsGroup(article.internalID, article.slug, section.internalID, index)
```
Comment (anandaroop) [inline src/app/Scenes/HomeView/Sections/HomeViewSectionArticlesCards.tsx:37]: **question:** If we want continuity with the old screen, do we want to follow this precedent?

https://github.com/artsy/eigen/blob/f179a02295479892cac6f0f379ae246711c5188d/src/app/Scenes/Articles/News/ArticlesCards.tsx#L21-L24

**Comment by dblandin** [comment]
@MounirDhahri I think this PR (and [the one in cohesion](https://github.com/artsy/cohesion/pull/526)), can be closed in favor of https://github.com/artsy/eigen/pull/10737.

As [discussed in Slack](https://artsy.slack.com/archives/C05EQL4R5N0/p1726146463686739), we can use `tappedArticleGroup` and distinguish between "Artsy Editorial" and "Market News" with ContextModule/OwnerType values.

**Pull request closed without merging**